### PR TITLE
Upgrade Fortune teller application to include resources using resource-config.json instead of running the Tracing agent.

### PR DESCRIPTION
**Pull request by olyagpl**
1. Upgrade fortune-maven and fortune-gradle applications to include resources using resource-config.json instead of running the Tracing agent. The fortune-maven can run on the latest GraalVM for JDK 23 release, unlike fortune-gradle, which requires JDK 21 for compatibility with Gradle.
With this upgrade, the native executable file size decreased significantly (even without -Os optimization!). For example, from 27M  to  13M  for fortune-maven. 

2. Delete staticfortune application which is basically a duplication of fortune-maven. It didn't show how to build a static image, but a mostly-static; there was no mention where this demo is used. We already have [Tiny Java Containers demos](https://github.com/graalvm/graalvm-demos/tree/master/tiny-java-containers) demonstrating Native Image linking options. 

Goals:
- to use the latest GraalVM for JDK 23 features (fortune-maven only, Gradle is not compatible with JDK 23)
- to promote the use of META-INF/native-image/resource-config.json
- to discourage using the Tracing agent

### CODE REVIEW

File: fortune-demo/README.md [hunk 1]
```diff
@@ -7,14 +7,11 @@ The Fortune demo is comprised of three sub-projects:
 - The _fortune-gradle_ project uses the [Gradle plugin for GraalVM Native Image building](https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html).
 - The _staticfortune_ project demonstrates how to build a statically linked native executable using the [Maven plugin for GraalVM Native Image building](https://graalvm.github.io/native-build-tools/latest/maven-plugin.html).
 
-This demo is a little more complicated than _HelloWorld_, and requires pre-configuration before building a native executable. The Maven and Gradle plugins for Native Image building can generate the required configuration for you by injecting the Java agent at package time.
-The plugin will also gather the diagnostic data at build time and write it to a dump file in the `target` directory.
-
 ## Preparation
 
-1. Download and install the latest GraalVM for JDK 21 using [SDKMAN!](https://sdkman.io/).
+1. Download and install GraalVM using [SDKMAN!](https://sdkman.io/).
```
Comment (bhoran) [inline fortune-demo/README.md:12]: Need to provide the link to the graalvm downloads for Windows users.

File: README.md [hunk 2]
```diff
@@ -53,7 +53,7 @@ cd graalvm-demos
     </tr>
     <tr>
       <td align="left" width="30%"><a href="/fortune-demo/">fortune-demo</a><br><a href="https://github.com/graalvm/graalvm-demos/actions/workflows/fortune-demo.yml"><img alt="fortune-demo" src="https://github.com/graalvm/graalvm-demos/actions/workflows/fortune-demo.yml/badge.svg" /></a></td>
-      <td align="left" width="70%">A fortune teller Unix program. Run it in JIT, build a native executable, or build a mostly-static native executable, using Gradle or Maven build tools.<br><strong>Technologies: </strong>Native Image, Native Build Tools Maven plugin <br><strong>Reference: </strong><a href="https://graalvm.github.io/native-build-tools/latest/maven-plugin-quickstart.html">Getting Started with Maven Plugin for GraalVM Native Image</a></td>
+      <td align="left" width="70%">A fortune teller Unix program. Build native executables using Gradle or Maven Native Image plugins.<br><strong>Technologies: </strong>Native Image, Native Build Tools Maven plugin <br><strong>Reference: </strong><a href="https://graalvm.github.io/native-build-tools/latest/maven-plugin-quickstart.html">Getting Started with Maven Plugin for GraalVM Native Image</a></td>
```
Comment (bhoran) [inline README.md:56]:
```suggestion
      <td align="left" width="70%">A fortune teller Unix program. Build a native executable using Gradle or Maven Native Image plugins.<br><strong>Technologies: </strong>Native Image, Native Build Tools Maven plugin <br><strong>Reference: </strong><a href="https://graalvm.github.io/native-build-tools/latest/maven-plugin-quickstart.html">Getting Started with Maven Plugin for GraalVM Native Image</a></td>
```
If this shows how to use Gradle, it should probably have a link to the Gradle plugin.

File: fortune-demo/README.md [hunk 3]
```diff
@@ -5,18 +5,15 @@ The demo is a Java program that simulates the traditional `fortune` Unix program
...
 - The _fortune-gradle_ project uses the [Gradle plugin for GraalVM Native Image building](https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html).
-- The _staticfortune_ project demonstrates how to build a statically linked native executable using the [Maven plugin for GraalVM Native Image building](https://graalvm.github.io/native-build-tools/latest/maven-plugin.html).
-
-This demo is a little more complicated than _HelloWorld_, and requires pre-configuration before building a native executable. The Maven and Gradle plugins for Native Image building can generate the required configuration for you by injecting the Java agent at package time.
-The plugin will also gather the diagnostic data at build time and write it to a dump file in the `target` directory.
 
 ## Preparation
 
-1. Download and install the latest GraalVM for JDK 21 using [SDKMAN!](https://sdkman.io/).
+1. Download and install GraalVM. The easiest way is with [SDKMAN!](https://sdkman.io/jdks#graal):
```
Comment (bhoran) [inline fortune-demo/README.md:11]:
```suggestion
1. Download and install GraalVM. The easiest way for Linux/macOS is with [SDKMAN!](https://sdkman.io/jdks#graal):
```

File: fortune-demo/README.md [hunk 4]
```diff
@@ -5,18 +5,15 @@ The demo is a Java program that simulates the traditional `fortune` Unix program
...
     ```bash
-    sdk install java 21.0.3-graal
+    sdk install java <version>-graal
     ```
-    
+    For other installation options, visit the [Downloads](https://www.graalvm.org/downloads/) section.
+
 2. Download or clone GraalVM demos repository:
```
Comment (bhoran) [inline fortune-demo/README.md:17]:
```suggestion
2. Download or clone the GraalVM demos repository:
```

File: fortune-demo/README.md [hunk 5]
```diff
@@ -29,104 +26,38 @@ The plugin will also gather the diagnostic data at build time and write it to a
...
-    The agent generates the configuration files in the _target/native/agent-output_ subdirectory.
-
-4. Build a native executable of this application with GraalVM Native Image and Maven:
-    ```bash
-    mvn -Pnative -Dagent package
-    ```
-    When the command completes, a native executable, `fortune`, is generated in the _target_ directory of the project and ready for use.
-
-5. Run the application by launching a native executable directly or with the Maven profile:
+    The command compiles the project, creates a JAR file with all dependencies, and then generates a native executable, `fortune`, in the _target_ directory, ready for use.
```
Comment (bhoran) [inline fortune-demo/README.md:33]:
```suggestion
    The command compiles the project, creates a JAR file with all dependencies, and then generates a native executable, `fortune`, in the _target/_ directory, ready for use.
```

File: fortune-demo/README.md [hunk 6]
```diff
@@ -29,104 +26,38 @@ The plugin will also gather the diagnostic data at build time and write it to a
...
-    ```bash
-    mvn -Pnative exec:exec@native
-    ```
+    The application will return a random saying.
 
 ## Fortune Gradle Demo
 
+> Requires GraalVM for JDK 21 for compatibility with Gradle. See [Gradle Java Compatibility Matrix](https://docs.gradle.org/current/userguide/compatibility.html).
+
 1. Change to the _fortune-demo/fortune-gradle_ directory:
```
Comment (bhoran) [inline fortune-demo/README.md:45]:
```suggestion
1. Change to the _fortune-demo/fortune-gradle/_ directory:
```

File: fortune-demo/README.md [hunk 7]
```diff
@@ -29,104 +26,38 @@ The plugin will also gather the diagnostic data at build time and write it to a
...
-    ./gradlew metadataCopy --task run --dir src/main/resources/META-INF/native-image
-    ```
-
-5. Build a native executable using configuration collected by the agent:
+2. Create a native executable using the [Gradle plugin for Native Image](https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html):
     ```bash
     ./gradlew nativeCompile
     ```
-
     When the command completes, a native executable, `fortune`, is generated in the _build/native/nativeCompile_ directory of the project and ready for use.
```
Comment (bhoran) [inline fortune-demo/README.md:54]:
```suggestion
    When the command completes, a native executable, `fortune`, is generated in the _build/native/nativeCompile/_ directory of the project and ready for use.
```

File: fortune-demo/README.md [hunk 8]
```diff
@@ -29,104 +26,38 @@ The plugin will also gather the diagnostic data at build time and write it to a
...
-    cd ../staticfortune
-    ```
-
-2. Build the project:
-    ```bash
-    mvn clean package
-    ```
 
-3. Run your application with the [agent](https://graalvm.github.io/native-build-tools/latest/maven-plugin.html#agent-support), on the JVM:
+3. Run the application from the native executable:
```
Comment (bhoran) [inline fortune-demo/README.md:56]:
```suggestion
3. Run the native executable:
```

File: fortune-demo/fortune-maven/pom.xml [hunk 9]
```diff
@@ -1,149 +1,77 @@
...
+
     <properties>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <native.maven.plugin.version>0.10.1</native.maven.plugin.version>
-        <junit.jupiter.version>5.8.1</junit.jupiter.version>
-        <maven.compiler.source>${java.specification.version}</maven.compiler.source>
-        <maven.compiler.target>${java.specification.version}</maven.compiler.target>
-        <imageName>fortune</imageName>
-        <mainClass>demo.Fortune</mainClass>
+        <maven.compiler.source>17</maven.compiler.source>
```
Comment (bhoran) [inline fortune-demo/fortune-maven/pom.xml:12]: The README says to use JDK21.

File: fortune-demo/fortune-gradle/fortune/src/main/java/demo/Fortune.java [hunk 10]
```diff
@@ -38,78 +38,41 @@
...
+ import java.io.InputStreamReader;
+ import java.util.ArrayList;
+ import java.util.Random;
+ import java.util.Scanner;
+ 
+ public class Fortune {
+ 
+     private static final String SEPARATOR = "%";
+     private static final Random RANDOM = new Random();
+     private ArrayList<String> fortunes = new ArrayList<>();
```
Comment (bhoran) [inline fortune-demo/fortune-gradle/fortune/src/main/java/demo/Fortune.java:54]: This should be `final`

File: fortune-demo/fortune-maven/src/main/java/demo/Fortune.java [hunk 11]
```diff
@@ -38,78 +38,41 @@
...
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Scanner;
 
 public class Fortune {
 
+    private static final String SEPARATOR = "%";
     private static final Random RANDOM = new Random();
-    private final ArrayList<String> fortunes = new ArrayList<>();
+    private ArrayList<String> fortunes = new ArrayList<>();
```
Comment (bhoran) [inline fortune-demo/fortune-maven/src/main/java/demo/Fortune.java:54]:
```suggestion
    private final ArrayList<String> fortunes = new ArrayList<>();
```

**Review by bhoran — CHANGES_REQUESTED** [review]

File: fortune-demo/fortune-gradle/fortune/src/main/java/demo/Fortune.java (hunk 10 above)
Comment (olyagpl) [reply fortune-demo/fortune-gradle/fortune/src/main/java/demo/Fortune.java:54]: Suggestion applied

**Review by bhoran — APPROVED** [review]

**Review by krisfoster — APPROVED** [review]
All looks good.

**Pull request merged**